gcp/compute/helpers.py

```diff
@@ -1,3 +1,36 @@
+def does_firewall_open_all_ports_to_any(firewall):
+    """
+    Returns True if firewall has a rule to open all ports to any source. Excludes ICMP.
+
+    >>> does_firewall_open_all_ports_to_any({})
+    False
+    >>> does_firewall_open_all_ports_to_any({'sourceRanges': ['1.1.1.1/1'], 'allowed': [{'ports': ['1', '2', '300']}]})
+    False
+    >>> does_firewall_open_all_ports_to_any({'sourceRanges': ['1.1.1.1/1'], 'allowed': [{'ports': ['0-65535']}]})
+    True
+    >>> does_firewall_open_all_ports_to_any({'sourceRanges': ['0.0.0.0/0'], 'allowed': [{'ports': ['0-65535']}]})
+    True
+    >>> does_firewall_open_all_ports_to_any({'sourceRanges': ['10.0.0.5/32'], 'allowed': [{'ports': ['0-65535']}]})
+    True
+    """
+    if does_firewall_open_all_ports_to_all(firewall):
+        return True
+
+    if firewall.get("sourceRanges") is None:
+        return False
+
+    for rule in firewall.get("allowed"):
+        if rule.get("IPProtocol", "") == "icmp":
+            continue
+        if not rule.get("ports"):
+            return True
+        for port_rule in rule.get("ports"):
+            if port_rule == "0-65535":
```

Comment:
does GCP normalize this to one range? e.g. could someone write `0-1,1-65535` and we wouldn't pick it up

Reply:
>does GCP normalize this to one range? e.g. could someone write 0-1,1-65535 and we wouldn't pick it up

GCP does not, but the default rules / docs point towards using `0-65535`. I think this is ok for now though. If any ports except 80 or 443 are opened to the public internet, `test_firewall_opens_any_ports_to_all` will catch it. We will miss big ranges allowed internally, but atm I think that's ok (and it's also true within our AWS tests)